internal/femto/authentication.go

```diff
@@ -0,0 +1,56 @@
+package femto
+
+import (
+	"errors"
+	"log/slog"
+	"net/http"
+)
+
+var (
+	NotAuthenticatedError   = errors.New("User does not have a valid authentication token")
+	InvalidCredientalsError = errors.New("Invalid credientals for creating an authentication token")
+)
+
+const (
+	AuthCookieField = "auth_cookie"
+)
+
+type AuthToken = string
+
+type Authenticator[T any] interface {
+	CreateToken(T) (AuthToken, error)
+	RevokeToken(token AuthToken) error
+	CheckToken(token AuthToken) bool
+}
+
+func AuthWrapGet[A any, T any](auth Authenticator[A], handle GetFunc[T]) GetFunc[T] {
+	return func(r *http.Request, l *slog.Logger) (T, error) {
+		var zero T
+		cookie, err := r.Cookie(AuthCookieField)
```

Comment:
Could we use HTTP Bearer authorization, instead of a cookie, as that's less browser specific?

Reply:
Im happy to do this, but I think we have to do a small rework on femto return types so I will delegate this to another PR later on. Paging @Euphoride : any opinions on Bearer?